### PR TITLE
Update Collapsible

### PR DESCRIPTION
**Pull request by willmcgugan**
- Fixed / updated docs for Collapsible
- Simplified implementation a bit

### CODE REVIEW

**Comment by willmcgugan** [comment]
@rodrigogiraoserrao you are most familiar with this...

File: docs/widgets/collapsible.md [hunk 1]
```diff
@@ -1,34 +1,37 @@
 # Collapsible
 
-!!! tip "Added in version 0.36"
+!!! tip "Added in version 0.37"
 
-Widget that wraps its contents in a collapsible container.
+A container with a title than can be used to show (expand) or hide (collapse) content, either by clicking or focusing and pressing ++enter++.
```
Comment (rodrigogiraoserrao) [inline docs/widgets/collapsible.md:5]:
```suggestion
A container with a title that can be used to show (expand) or hide (collapse) content, either by clicking or focusing and pressing ++enter++.
```

File: docs/widgets/collapsible.md [hunk 2]
```diff
@@ -1,34 +1,37 @@
...
+
+Here is an example of using the constructor to add content:
 
 ```python
 def compose(self) -> ComposeResult:
     yield Collapsible(Label("Hello, world."))
 ```
 
-Alternatively, you can compose other widgets under the context manager:
+Here's how the to use it with the context manager.
```
Comment (rodrigogiraoserrao) [inline docs/widgets/collapsible.md:22]: For consistency with the example above.

```suggestion
Here's how the to use it with the context manager:
```

File: src/textual/widgets/_collapsible.py [hunk 3]
```diff
@@ -1,87 +1,93 @@
...
+    def action_toggle(self) -> None:
+        """Toggle the state of the parent collapsible."""
+        self.post_message(self.Toggle())
+
+    def render(self) -> RenderableType:
+        """Compose right/down arrow and label."""
+        if self.collapsed:
+            return Text.assemble(self.collapsed_symbol, " ", self.label)
+        else:
+            return Text.assemble(self.expanded_symbol, " ", self.label)
```
Comment (rodrigogiraoserrao) [inline src/textual/widgets/_collapsible.py:73]: Doesn't this get rerun every single time we need to show the title, even if the collapsed state didn't change?
Couldn't we cache this if we inherited from `Static`?

File: src/textual/widgets/_collapsible.py [hunk 4]
```diff
@@ -1,87 +1,93 @@
...
-        Title {
-            width: 100%;
-            height: auto;
-        }
+    BINDINGS = [Binding("enter", "toggle", "Toggle collapsible", show=False)]
 
-        Title:hover {
-            background: grey;
-        }
+    collapsed = reactive(False)
```
Comment (rodrigogiraoserrao) [inline src/textual/widgets/_collapsible.py:40]: I find it a bit odd that `Collapsible.collapsed` starts off as `True` and this one starts off as `False`!

**Review by rodrigogiraoserrao — CHANGES_REQUESTED** [review]
I think I spotted something small that needs changing.

File: src/textual/widgets/_collapsible.py (hunk 4 above)
Comment (willmcgugan) [reply src/textual/widgets/_collapsible.py:40]: Is a bit. Changed to match the default.

File: src/textual/widgets/_collapsible.py (hunk 3 above)
Comment (willmcgugan) [reply src/textual/widgets/_collapsible.py:73]: Only when it is repainted. It's such a trivial function that caching the renderable may not even be a win.

But I have noticed, that it is doing a little more work that required!

**Pull request merged**